tests/check/check.go

```diff
@@ -1239,6 +1258,84 @@ condition xcond(x: string) {
+						objectType, _ := tuple.SplitObject(assertion.Tuple.GetObject())
+						resp, err := client.ListObjects(ctx, &openfgav1.ListObjectsRequest{
+							StoreId:              storeID,
+							AuthorizationModelId: modelID,
+							Type:                 objectType,
+							User:                 assertion.Tuple.GetUser(),
+							Relation:             assertion.Tuple.GetRelation(),
+							ContextualTuples: &openfgav1.ContextualTupleKeys{
+								// TODO
+								TupleKeys: []*openfgav1.TupleKey{},
+							},
+							Context: assertion.Context,
+						})
+
+						if assertion.ListObjectsErrorCode != 0 {
+							require.Error(t, err)
+							e, ok := status.FromError(err)
+							require.True(t, ok)
+							require.Equal(t, assertion.ListObjectsErrorCode, int(e.Code()))
+							return
+						}
+						if assertion.ErrorCode == 0 {
+							require.NoError(t, err)
+							if assertion.Expectation {
+								// >=, not >, because some users have access to more than one object
+								require.GreaterOrEqual(t, len(resp.GetObjects()), 1, "at least one object should be returned")
+								require.Contains(t, resp.GetObjects(), assertion.Tuple.GetObject(), "object should be in the list")
+							} else {
+								require.NotContains(t, resp.GetObjects(), assertion.Tuple.GetObject(), "object should not be in the list")
+							}
+						}
+					})
+					t.Run(fmt.Sprintf("assertion_list_users_%s", assertion.Name), func(t *testing.T) {
+						objectType, objectID := tuple.SplitObject(assertion.Tuple.GetObject())
+						userObject, userRelation := tuple.SplitObjectRelation(assertion.Tuple.GetUser())
+						userObjectType, _ := tuple.SplitObject(userObject)
+						relation := assertion.Tuple.GetRelation()
+						resp, err := client.ListUsers(ctx, &openfgav1.ListUsersRequest{
+							StoreId:              storeID,
+							AuthorizationModelId: modelID,
+							Object:               &openfgav1.Object{Type: objectType, Id: objectID},
+							Relation:             relation,
+							UserFilters:          []*openfgav1.UserTypeFilter{{Type: userObjectType, Relation: userRelation}},
+							ContextualTuples:     []*openfgav1.TupleKey{}, // TODO
+							Context:              assertion.Context,
+						})
+
+						if assertion.ListUsersErrorCode != 0 {
+							require.Error(t, err)
+							e, ok := status.FromError(err)
+							require.True(t, ok)
+							require.Equal(t, assertion.ListUsersErrorCode, int(e.Code()))
+							return
+						}
+
+						if assertion.ErrorCode == 0 {
+							require.NoError(t, err)
+							responseUsers := make([]string, len(resp.GetUsers()))
+							wildcardUserPresent := false
+							for i, u := range resp.GetUsers() {
+								responseUsers[i] = tuple.UserProtoToString(u)
+								if strings.HasSuffix(responseUsers[i], ":*") {
+									wildcardUserPresent = true
+								}
+							}
+							if assertion.Expectation {
+								if wildcardUserPresent {
```

Comment:
Verify that it is the intention whenever we expect a wildcard user AND expectation == true, we will not be doing assertion for list user.

Reply:
So the scenario is that ListUsers response is `users:[user:*, .....]`. Then, we don't want to do the `is user:target part of the response?` assertion because it will fail.

Reply:
agree.  Can you add comment to make it easier to note that?